lib/puppet/parser/resource.rb

```diff
@@ -199,6 +199,51 @@ def raw_tagged?(tag_array)
     super || ((scope_resource = scope.resource) && !scope_resource.equal?(self) && scope_resource.raw_tagged?(tag_array))
   end
 
+  # Fill resource params from a capability
+  #
+  # This backs 'consumes => Sql[one]'
+  def add_parameters_from_consume
+    return unless self[:consume]
```

Comment:
this should probably be `return if self[:consumes].nil?` to avoid being screwed if code managed to get a boolean false in there.